build.gradle

```diff
@@ -137,7 +122,7 @@ task copyServerSelectorResources(type: Copy, dependsOn: npm_install) {
 	into('src/main/resources/server-selector')
 }
 
-task convertLocalizationEncoding() {
+tasks.register('convertLocalizationEncoding') {
```

Comment:
since this change seems unrelated to the switch to another gradle plugin, could we do it on the main branch first to reduce the delta?

Same for the other minor syntax changes below, e.g.
`from components.java`
vs
`from(components["java"])`
etc. To reduce the risk and make it easier to review, all unrelated improvements should be extracted and be done in the main branch first.